GraphConsoleAppV3/GraphRequests.cs

```diff
@@ -0,0 +1,1551 @@
+            }
+            catch (Exception e)
+            {
+                Program.WriteError("\nError getting TenantDetails {0}", Program.ExtractErrorMessage(e));
+            }
+
+            if (tenant == null)
+            {
+                Console.WriteLine("Tenant not found");
+                return null;
+            }
+            else
+            {
+                TenantDetail tenantDetail = (TenantDetail) tenant;
+                Console.WriteLine("Tenant Display Name: " + tenantDetail.DisplayName);
+
+                // Get the Tenant's Verified Domains 
+                var initialDomain = tenantDetail.VerifiedDomains.First(x => x.Initial.HasValue && x.Initial.Value);
+                Console.WriteLine("Initial Domain Name: " + initialDomain.Name);
+                var defaultDomain = tenantDetail.VerifiedDomains.First(x => x.@default.HasValue && x.@default.Value);
+                Console.WriteLine("Default Domain Name: " + defaultDomain.Name);
+
+                // Get Tenant's Tech Contacts
+                foreach (string techContact in tenantDetail.TechnicalNotificationMails)
+                {
+                    Console.WriteLine("Tenant Tech Contact: " + techContact);
+                }
+                return tenantDetail;
+            }
+
+            #endregion
+        }
+
+        private static async Task<User> GetSignedInUser(ActiveDirectoryClient client)
+        {
+            #region Get signed user info, get their photo, and update their photo
+
+            #region Get signed in user details
+
+            User signedInUser = new User();
+            try
+            {
+                signedInUser = (User) await client.Me.ExecuteAsync();
+                Console.WriteLine("\nUser UPN: {0}, DisplayName: {1}", signedInUser.UserPrincipalName,
+                    signedInUser.DisplayName);
+            }
+            catch (Exception e)
+            {
+                Program.WriteError("\nError getting signed in user {0}", Program.ExtractErrorMessage(e));
+            }
+
+
+            #endregion
+
+            #region get signed in user's photo
+
+            if (signedInUser.ObjectId != null)
+            {
+                IUser sUser = (IUser) signedInUser;
+                IStreamFetcher photo = (IStreamFetcher) sUser.ThumbnailPhoto;
+                try
+                {
+                    DataServiceStreamResponse response =
+                        await photo.DownloadAsync();
+                    Console.WriteLine("\nUser {0} GOT thumbnailphoto", signedInUser.DisplayName);
+                }
+                catch (Exception e)
+                {
+                    Program.WriteError("\nError getting the user's photo - may not exist {0}",
+                        Program.ExtractErrorMessage(e));
+                }
+            }
+            return signedInUser;
+
+            #endregion
+        }
+
+        private static async Task UpdateSignedInUsersPhoto(IUser signedInUser)

```

Comment:
changed

---
In reply to: [81251182](https://github.com/Azure-Samples/active-directory-dotnet-graphapi-console/pull/37#discussion_r81251182) [](ancestors = 81251182)